### ISSUE TITLE
[REQUEST] GitHub Actions for Multi-Architecture Docker Image Builds

### ISSUE THREAD

**Issue by 0x4139** [theroyallab/tabbyAPI#157]
# Proposal: 

**Is your feature request related to a problem? Please describe.**

Currently, the project does not have automated workflows for building Docker images for multiple architectures. This limitation causes frustration when trying to support diverse deployment environments, leading to manual builds and potential inconsistencies.

**Describe the solution you'd like**

I propose adding GitHub Actions workflows to automate the building and pushing of Docker images for various architectures (e.g., x86, ARM). This will streamline our CI/CD pipeline, ensuring consistent and reliable builds across different platforms.

**Describe alternatives you've considered**

- **Manual Builds**: Continue building Docker images manually for each architecture, but this is time-consuming and prone to human error.
- **Third-Party CI Services**: Use third-party CI services for multi-architecture builds, but this adds complexity and potential cost.

**Why should this feature be added?**

- **Cross-Platform Compatibility**: Ensures support for diverse environments like cloud, on-premises, and edge devices.
- **Consistency and Reliability**: Automated builds maintain consistent build conditions, reducing human error.
- **Improved CI/CD Pipeline**: Enables faster delivery of updates and new features through continuous integration and deployment.
- **Community Support**: Simplifies testing and deployment for contributors, encouraging more contributions.
- **Future-Proofing**: Adapts to evolving hardware landscapes by supporting multiple architectures.

**Examples**

- **Current State**: Manually building and pushing Docker images for different architectures can lead to inconsistencies and delays.
- **Proposed State**: With GitHub Actions, Docker images for `linux/amd64` and `linux/arm64` are automatically built and pushed, ensuring consistency and reducing manual effort.

**Additional context**

Here is an example of the GitHub Actions workflow that can be added:

```yaml
name: Build and Push Docker Images

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v1

    - name: Login to Docker Hub
      uses: docker/login-action@v1
      with:
        username: ${{ secrets.DOCKER_USERNAME }}
        password: ${{ secrets.DOCKER_PASSWORD }}

    - name: Build and push Docker images
      uses: docker/build-push-action@v2
      with:
        context: .
        push: true
        platforms: linux/amd64,linux/arm64
        tags: user/repo:latest
```

Thank you for considering this proposal. Please feel free to reach out for any clarifications or further discussions, I am more than happy to submit a pull request.

---

**Comment by bdashore3** [comment]
Can be linked to #123. A pull request would be helpful since I don't use docker.

**Comment by bdashore3** [comment]
Resolved via #171
